### PR TITLE
[MOB-9620] unit test updates for identity resolution combinations

### PR DESCRIPTION
**Pull request by evantk91**
## 🔹 Jira Ticket(s) if any

* [MOB-9620](https://iterable.atlassian.net/browse/MOB-9620)

## ✏️ Description

> This pull request adds unit tests to cover all the combinations of the identity resolution parameter.

[MOB-9620]: https://iterable.atlassian.net/browse/MOB-9620?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: iterableapi/src/test/java/com/iterable/iterableapi/IterableApiMergeUserEmailTests.java [hunk 1]
```diff
@@ -235,72 +253,96 @@ public void testCriteriaNotMetUserIdDefault() throws Exception {
...
 
     @Test
-    public void testCriteriaMetUserIdMergeFalse() throws Exception {
+    public void testCriteriaMetUserIdDefault() throws Exception {
         while (server.takeRequest(1, TimeUnit.SECONDS) != null) { }
         addResponse(IterableConstants.ENDPOINT_TRACK_ANON_SESSION);
+        addResponse(IterableConstants.ENDPOINT_MERGE_USER);
         triggerTrackPurchaseEvent("test", "keyboard", 4.67, 3);
         shadowOf(getMainLooper()).idle();
         assertEquals("", getEventData());
```
Comment (Ayyanchira) [inline iterableapi/src/test/java/com/iterable/iterableapi/IterableApiMergeUserEmailTests.java:275]: If we triggerTrackEventPurchase, should the getEventData not give us some value? 
I am assuming getEventData is reading from internal storage.

**Review by Ayyanchira — COMMENTED** [review]
Made some comments. Later half of criteria matched will review once I clarify what happens with getEventData()

File: iterableapi/src/test/java/com/iterable/iterableapi/IterableApiMergeUserEmailTests.java [hunk 2]
```diff
@@ -182,51 +182,69 @@ private void addResponse(String endPoint) {
...
+        assertEquals(userId, IterableApi.getInstance().getUserId());
+        assertEquals("", getEventData());
+    }
+
+    @Test
+    public void testCriteriaNotMetUserIdReplayTrueMergeFalse() throws Exception {
+        while (server.takeRequest(1, TimeUnit.SECONDS) != null) { }
+        addResponse(IterableConstants.ENDPOINT_TRACK_ANON_SESSION);
+        triggerTrackPurchaseEvent("test", "keyboard", 5, 1);
+        shadowOf(getMainLooper()).idle();
```
[the remark this answers is not in the document]
Comment (evantk91) [reply iterableapi/src/test/java/com/iterable/iterableapi/IterableApiMergeUserEmailTests.java:208]: I will open a new ticket for this and PR.

**Comment by evantk91** [comment]
Oops I accidently merged through fork. I will address your comments in a separate branch.

**Pull request merged**